### PR TITLE
Make profile card glow based on device orientation

### LINKED ISSUE #210: Make profile card glow based on device orientation
## Description
To provide an experience similar to holding a physical card, implement a feature that makes the profile card glow based on the device's orientation using the gyroscope sensor.

## Links
Design: https://www.figma.com/design/XUk8WMbKCeIdWD5cz9P9JC/DroidKaigi-2024-App-UI

### PR DESCRIPTION
**Pull request by shunm-999**
## Issue
- close #210

## Overview (Required)
- Implemented a feature that makes the profile card glow based on the device's orientation using the gyroscope sensor.
- To avoid making the QR code difficult to recognize, the implementation is applied only to the front.

## Links
- 

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Before | After
:--: | :-
 | 

## Movie (Optional)
Before | After
:--: | :--:
<video src="" width="300" > | <video src="https://github.com/user-attachments/assets/85cdfe16-b910-4d03-9d52-56dc407889f4" width="300" >

### CODE REVIEW

File: core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/OrientationSensorManager.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+internal interface OrientationSensorManager {
+    val onOrientationChanged: (Orientation) -> Unit
+    fun start()
+    fun stop()
+}
+
+@Composable
+internal expect fun getOrientationSensorManager(
```
Comment (takahirom) [inline core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/OrientationSensorManager.kt:22]: Beautiful API!

File: feature/profilecard/src/commonMain/kotlin/io/github/droidkaigi/confsched/profilecard/HologramaticEffectModifier.kt [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    override fun InspectorInfo.inspectableProperties() {
+        name = "HologramaticEffectNode"
+        properties["orientation"] = orientation
+    }
+}
+
+private data class HologramaticEffectNode(
+    var orientation: Orientation,
+) :
+    Modifier.Node(),
```
Comment (takahirom) [inline feature/profilecard/src/commonMain/kotlin/io/github/droidkaigi/confsched/profilecard/HologramaticEffectModifier.kt:44]: Modifier.Node! 👍

File: core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/OrientationSensorManager.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import androidx.compose.runtime.Composable
+
+data class Orientation(
+    val azimuth: Float,
+    val pitch: Float,
+    val roll: Float,
+) {
+    companion object {
+        val ZERO = Orientation(0f, 0f, 0f)
```
Comment (takahirom) [inline core/ui/src/commonMain/kotlin/io/github/droidkaigi/confsched/ui/OrientationSensorManager.kt:11]: We are using PascalCase for Compose constants. Could you check it out?
https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#singletons-constants-sealed-class-and-enum-class-values

File: feature/profilecard/src/commonMain/kotlin/io/github/droidkaigi/confsched/profilecard/HologramaticEffectModifier.kt [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                        drawRect(
+                            topLeft = Offset(
+                                -size.width * 0.5f,
+                                -size.height * 0.5f,
+                            ),
+                            size = Size(
+                                size.width * 2f,
+                                size.height * 2f,
+                            ),
+                            brush = Brush.linearGradient(
```
Comment (takahirom) [inline feature/profilecard/src/commonMain/kotlin/io/github/droidkaigi/confsched/profilecard/HologramaticEffectModifier.kt:73]: 👍

**Review by takahirom — APPROVED** [review]
This code showcases your expertise! Thank you for your awesome contribution!
I've added a few minor comments, but overall, I think we can merge this.

**Comment by takahirom** [comment]
I've addressed the points I commented on. Let's merge.

**Comment by shunm-999** [comment]
> I've addressed the points I commented on. Let's merge.

Thank you for your commits!

**Pull request merged**